src/CTA.Rules.Config/Utils.cs

```diff
@@ -191,20 +192,52 @@ public static string ThreadSafeExportStringToFile(string filePath, string conten
             }
         }
 
+
+        public static string CopySolutionToDestinationLocation(string solutionName, string solutionPath)
+        {
+            string slnDirPath = Directory.GetParent(solutionPath).FullName;
+            string root = Path.GetPathRoot(slnDirPath);
+            string relativeSrc = Path.GetRelativePath(root, slnDirPath);
+            
+            var newTempDirPath = Path.Combine(Directory.GetParent(slnDirPath).FullName, Guid.NewGuid().ToString());
+            string destPath = Path.Combine(newTempDirPath, relativeSrc);
+            string newSlnPath = CopyFolderToTemp(solutionName, slnDirPath, destPath);
+
+
+            if (solutionPath.Contains(".sln") && File.Exists(solutionPath))
+            {
+                IEnumerable<string> projects = GetProjectPathsFromSolutionFile(solutionPath);
+                foreach (string project in projects)
+                {
+                    string projPath = Directory.GetParent(project).FullName;
+                    relativeSrc = Path.GetRelativePath(solutionPath, projPath);
+                    string projName = Path.GetFileName(project);
+
+                    if (!IsSubPathOf(slnDirPath, projPath))
+                    {
+                        string relDestPath = Path.Combine(destPath, relativeSrc);
+                        CopyFolderToTemp(projName, projPath, relDestPath);
+                    }
+                }
+            }
+            return newSlnPath;
+        }
```

Comment:
Similar changes were made to codelyzer: 
https://github.com/aws/codelyzer/pull/127/files
https://github.com/aws/codelyzer/pull/139/files

Can we reuse that code?

Reply:
we can reuse but codelyzer is currently targeting .net6. Need to reference utils once CTA also get framework upgrade

Reply:
We should (fingers crossed) be able to finish the CTA .NET 6 upgrade once this codelyzer PR is merged: https://github.com/aws/codelyzer/pull/141